src/core/services/layout/layout.js

```diff
@@ -367,6 +375,30 @@
   }
 
   /**
+   * Centralize warnings for known flexbox issues (especially IE-related issues)
+   */
+  function validateAttributeUsage(className, attr, element, $log){
```

Comment:
The problem is that it adds a run-time cost to every instance of a layout directive. It is small, but it could add up (entirely speculation). 

(One of the things I want to do for Angular 2 is have code blocks that we can run in dev mode but strip for production.)